src/localPRG.cpp

```diff
@@ -247,54 +247,30 @@ std::vector<uint32_t> LocalPRG::build_graph(const Interval &i,
     return end_ids;
 }
 
-std::vector<Path> LocalPRG::shift(Path p) const {
+std::vector<Path> LocalPRG::shift(Path path) const {
     // returns all paths of the same length which have been shifted by one position along prg graph
-    Path q;
-    q = p.subpath(1, p.length() - 1);
-    std::vector<LocalNodePtr> n;
-    std::vector<Path> return_paths;
-    std::deque<Path> short_paths = {q};
-    std::vector<Path> k_paths;
-    bool non_terminus;
-    //uint exp_num_return_seqs = 0;
-
-    // first find extensions of the path
-    while (!short_paths.empty()) {
-        p = short_paths.front();
-        n = nodes_along_path(p);
-        short_paths.pop_front();
-
-        // if we can extend within the same localnode, do
-        if (p.get_end() < n.back()->pos.get_end()) {
-            p.path.back().length += 1;
-            k_paths.push_back(p);
-        } else if (p.get_end() != (--(prg.nodes.end()))->second->pos.get_end()) {
-            for (uint32_t i = 0; i != n.back()->outNodes.size(); ++i) {
-                //exp_num_return_seqs += 1;
-                short_paths.push_back(p);
-                short_paths.back().add_end_interval(
-                        Interval(n.back()->outNodes[i]->pos.start, n.back()->outNodes[i]->pos.start));
-            }
-        }
-    }
+    Path truncated_path = path.subpath(1, path.length() - 1);
```

Comment:
Does this line shift or truncate the path? If truncate, why?

Reply:
This line takes the truncated path from position 1 along the original path to the end. I want the paths which are shifted along one position in the graph. I have the first `k-1` bases length of this desired path(s) already, so I use them and look for the next `1` base length to add to the end of the path